src/util/synth.ts

```diff
@@ -49,6 +49,14 @@ export interface DirectorySnapshotOptions {
    * @default [] include all files
    */
   readonly excludeGlobs?: string[];
+
+  /**
+   * Only read the names of files and not the contents.
+   * Value of `[path]` will be `true` if it exists.
+   *
+   * @default false include file content
+   */
+  readonly excludeContent?: boolean;
```

Comment:
optional: The type signature is different from "excludeGlobs" so it might make sense to name it differently
```suggestion
  /**
   * Only snapshot the names of files and not their contents.
   * The value for a path will be `true` if it exists.
   *
   * @default false include file content
   */
  readonly omitContents?: boolean;
```